msticpy/data/uploaders/splunk_uploader.py

```diff
@@ -40,12 +41,28 @@ def __init__(self, username: str, host: str, password: str, **kwargs):
 
     def connect(self):
         """Connect to Splunk host."""
-        self.driver.connect(
-            host=self.workspace,
-            username=self.user,
-            password=self.workspace_secret,
-            port=self.port,
-        )
+        if self._kwargs:
+            if self.username:
+                self.driver.connect(
+                    host=self.host,
+                    username=self.username,
+                    password=self.password,
+                    port=self.port,
+                )
+            elif self.splunkToken:
+                self.driver.connect(
+                    host=self.host,
+                    splunkToken=self.splunkToken,
+                    port=self.port,
+                )
+            else:
+                print("username and splunkToken are missing.")
```

Comment:
I'm not a big fan of printing out stuff without any user control.
It might better to add a logger to the module and output the message as an info message.
```python3
import logging
...
logger = logging.getLogger(__name__)

# then in the code 
logger.info("username/token not given as parameters - using config settings.")
```

Reply:
Checked and Done!